packages/google_maps_flutter/google_maps_flutter_android/lib/src/google_maps_flutter_android.dart

```diff
@@ -1219,6 +1219,37 @@ PlatformJointType platformJointTypeFromJointType(JointType jointType) {
   return PlatformJointType.mitered;
 }
 
+/// Converts platform interface's Cap to Pigeon's PlatformCap.
+@visibleForTesting
+PlatformCap platformCapFromCap(Cap cap) {
+  // TODO(schectman): Convert Cap to structured data.
+  // https://github.com/flutter/flutter/issues/155121
+  final List<Object> json = cap.toJson() as List<Object>;
+  final String tag = json[0] as String;
+  switch (tag) {
+    case 'buttCap':
+      return PlatformCap(type: PlatformCapType.buttCap);
+    case 'roundCap':
+      return PlatformCap(type: PlatformCapType.roundCap);
+    case 'squareCap':
+      return PlatformCap(type: PlatformCapType.squareCap);
+    case 'customCap':
+      final Object bitmapDescriptor = json[1];
+      final double refWidth = json[2] as double;
+      return PlatformCap(
+          type: PlatformCapType.customCap,
+          bitmapDescriptor: bitmapDescriptor,
+          refWidth: refWidth);
+  }
+  // The string tags used to identify the type of cap comes from a different
+  // package, which could get a new value at
+  // any time, so provide a fallback that ensures this won't break when used
+  // with a version that contains new values. This is deliberately outside
+  // the switch rather than a `default` so that the linter will flag the
+  // switch as needing an update.
```

Comment:
This last sentence does not apply in this context. String-value enums cannot be exhaustive, so the linter will never flag this.